### PR TITLE
Handle mismatch between questionnaire and questionnaire response

### PR DESCRIPTION
**Pull request by nikhil-ht**
**Overview**

- Handle mismatch between questionnaire and questionnaire response

**Test**

- Tested by adding  new item  in questionnaire Response which is not present in questionnaire

### CODE REVIEW

**Review by semicolin — COMMENTED** [review]
It looks like this only handles the case where QR has extra items. We also want to handle the case where QR is missing items, so the form will show empty/default values for those items instead of throwing error. Does it make sense to add that here, or should that be a separate PR?

Also, should it log a warning when there is a mismatch?

**Comment by nikhil-ht** [comment]
>

**Comment by nikhil-ht** [comment]
> It looks like this only handles the case where QR has extra items. We also want to handle the case where QR is missing items, so the form will show empty/default values for those items instead of throwing error. Does it make sense to add that here, or should that be a separate PR?
> 
> Also, should it log a warning when there is a mismatch?


@semicolin  If the qr does not contain the item that is present in the questionnaire, it shows an empty field by default.

will add a log for mismatch

**Comment by nikhil-ht** [comment]
Tested below  scenario :

**Scenario 1: If the QR has  item that not present in Questionnaire** 
scenario 1

**Scenario 2: If the questions not present in QR**

scenario 2

**Scenario 3: If the answer of question  not present in QR**

scenario 3

File: src/fhir/sdc-import-common.js [hunk 1]
```diff
@@ -885,7 +885,24 @@ function addCommonSDCImportFns(ns) {
    * @param qr a QuestionnaireResponse instance
    * @returns {{}} an updated LForms form definition, with answer data
    */
-  qrImport.mergeQuestionnaireResponseToLForms = function(formData, qr) {
+  let mismatchItem = [];
```
Comment (semicolin) [inline src/fhir/sdc-import-common.js]: I think this should be declared inside function scope. It's not used outside the function, right?

File: src/fhir/sdc-import-common.js [hunk 2]
```diff
@@ -885,7 +885,24 @@ function addCommonSDCImportFns(ns) {
...
+
+      if (!matched) {
+        mismatchItem.push(obj1.linkId);
+      }
+      return matched;
+      //Handle the missmatch between questionnaire and questionnaireResponse 
+      //Keep only those item in qr whoes present in  formData
+    });
+
+    console.log('Mismatched items linkId:', mismatchItem);
```
Comment (semicolin) [inline src/fhir/sdc-import-common.js]: we should not print unnecessary log message when there are no mismatched items.

**Review by semicolin — CHANGES_REQUESTED** [review]

File: src/fhir/sdc-import-common.js (hunk 2 above)
Comment (nikhil-ht) [reply src/fhir/sdc-import-common.js]: Okay @semicolin  will handle this situation

File: src/fhir/sdc-import-common.js (hunk 1 above)
Comment (nikhil-ht) [reply src/fhir/sdc-import-common.js]: Yes @semicolin  will move this is in function scope

**Review by semicolin — APPROVED** [review]

**Review by yashpmohod — APPROVED** [review]

**Pull request merged**